src/main/java/org/swisspush/redisques/handler/RedisquesHttpRequestHandler.java

```diff
@@ -547,90 +506,80 @@ private void setConfiguration(RoutingContext ctx) {
-                        resultObject.put(QUEUES, updatedQueuesList);
-                        jsonResponse(ctx.response(), resultObject);
-                    });
-                } else {
-                    // there was no result, we as well return an empty result
-                    JsonObject resultObject = new JsonObject();
-                    resultObject.put(QUEUES, new JsonArray());
-                    jsonResponse(ctx.response(), resultObject);
-                }
-            } else {
-                String error = "Error gathering names of active queues";
-                log.error(error, reply.cause());
-                respondWith(StatusCode.INTERNAL_SERVER_ERROR, error, ctx.request());
+    private class MyQueueStatsMentor implements GetQueueStatsMentor<RoutingContext> {
+
+        @Override
+        public boolean includeEmptyQueues(RoutingContext ctx) {
+            return evaluateUrlParameterToBeEmptyOrTrue(EMPTY_QUEUES_PARAM, ctx.request());
+        }
+
+        @Override
+        public int limit(RoutingContext ctx) {
+            return extractLimit(ctx);
+        }
+
+        @Override
+        public String filter(RoutingContext ctx) {
+            return ctx.request().params().get(FILTER);
+        }
+
+        @Override
+        public void onQueueStatistics(List<QueueStatsService.Queue> queues, RoutingContext ctx) {
+            var rsp = ctx.request().response();
+            rsp.putHeader(CONTENT_TYPE, APPLICATION_JSON);
+            rsp.setChunked(true);
+            rsp.write("{\"queues\": [");
+            JsonObject queueJson = new JsonObject();
+            boolean isFirst = true;
+            for (QueueStatsService.Queue queue : queues) {
+                rsp.write(isFirst ? "" : ",");
+                isFirst = false;
+                queueJson.clear();
+                queueJson.put(MONITOR_QUEUE_NAME, queue.getName());
+                queueJson.put(MONITOR_QUEUE_SIZE, queue.getSize());
+                // TODO old impl did bloat result with empty strings for whatever undocumented
+                //      reason. Those fields should be set to 'null' (or we could even skip them
+                //      entirely in JSON), as obviously the information is not available. But
+                //      we'll add the same bloat as we don't know if any downstream code now
+                //      relies on this behavior.
+                Long epochMs;
+                epochMs = queue.getLastDequeueAttemptEpochMs();
+                queueJson.put("lastDequeueAttempt", epochMs == null ? "" : asUglyDate(epochMs));
+                epochMs = queue.getLastDequeueSuccessEpochMs();
+                queueJson.put("lastDequeueSuccess", epochMs == null ? "" : asUglyDate(epochMs));
+                epochMs = queue.getNextDequeueDueTimestampEpochMs();
+                queueJson.put("nextDequeueDueTimestamp", epochMs == null ? "" : asUglyDate(epochMs));
+                rsp.write(queueJson.encode());
```

Comment:
I'm not sure how versioning is handled here, but we might want to remember to remove these for the next major (breaking) release.

Reply:
I don't get it. What do you refer to with "*these*"? Do you mean what I wrote in the TODO?